apps/api/src/lib/id-generation/id-generation.module.ts

```diff
@@ -0,0 +1,29 @@
+import { Module } from '@nestjs/common';
+import { IdGenerationController } from '../../app/controllers/id-generation/id-generation.controller';
+import { AggregateId } from '../../domain/types/AggregateId';
+import { ArangoConnectionProvider } from '../../persistence/database/arango-connection.provider';
+import { DatabaseProvider } from '../../persistence/database/database.provider';
+import { PersistenceModule } from '../../persistence/persistence.module';
+import { ArangoIdRepository } from '../../persistence/repositories/arango-id-repository';
+import { IdManagementService } from './id-management.service';
+import { IIdRepository } from './interfaces/id-repository.interface';
+
+@Module({
+    imports: [PersistenceModule],
+    providers: [
+        {
+            provide: 'ID_REPOSITORY',
```

Comment:
Where is this dependency used?

Reply:
It's injected in the `IdManagementService`. We should think twice about this. We want to make sure that by simply swapping out the `RepositoryProvider` every dependency gets updated instantly (e.g. if you want to replace Arango or if you want to do an in-memory or mock).